### DISCUSSION TITLE
FastAPI and MongoEngine integration : Errors

### DISCUSSION THREAD

**Discussion by naheedroomy** [fastapi/fastapi#6606] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
Database connection string:

from mongoengine import connect
connect(host="mongodb://127.0.0.1:27017/pythonmongodb")


My models:
class User(BaseModel):
    id: Optional[str]
    username: str
    email: str
    password: str


class Users(Document):
    username: StringField(required=True)
    password = StringField(required=True, min_length=4)
    email = EmailField(required=True, unique=True)



Post request to create user:

@user.post('/users', response_model=User, tags=["users"])
async def create_user(user: User):
    new_user = Users(**user.dict()).save()
    id = new_user.id

    return id
```

### Description

Hey guys, was developing on Flask for a while before I came across FastAPI and absolutely love it.

I have a small question though. I very much prefer MongoEngine over other modules like PyMongo and Motor because of its ORM like nature and I would prefer to have an answer related to MongoEngine.

So this is my question. I've connected my simple user management application to a db.
I've also defined the model of the user, and the document model as well.
Now, I'm trying to add a user to this, however am getting the error stating
mongoengine.errors.FieldDoesNotExist: The fields "{'username'}" do not exist on the document "Users"
As for my database, I do have a database called "pythonmongodb" which has a collection called "Users

I'm still new to this whole API thing, so please do advice me in what you think I'm doing wrong!

### Operating System

Linux

### Operating System Details

Ubuntu 20.04 LTS

### FastAPI Version

0.68.1

### Python Version

3.9.7

**Comment by Kludex** [comment]
Thanks for coming back and closing it! It's much appreciated :)
